### PR TITLE
Expose IsGenericTypeParameter and IsGenericMethodParameter on System.…

### PR DESCRIPTION
**Pull request by AtsushiKan**
…Type

Api approved here:
  https://github.com/dotnet/corefx/issues/23883

### CODE REVIEW

**Comment by AtsushiKan** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by AtsushiKan** [comment]
@dotnet-bot test Linux x64 Release Build

File: src/System.Runtime/tests/System/Reflection/SignatureTypes.netcoreapp.cs [hunk 1]
```diff
@@ -491,7 +500,16 @@ private static void TestSignatureTypeInvariants(Type type)
 
                 int position = type.GenericParameterPosition;
                 Assert.True(position >= 0);
-                Assert.Equal("!!" + position, type.Name);
+
+                if (type.IsGenericTypeParameter)
+                {
+                    throw new Exception("Unexpected: There is no mechanism at this time to create Signature Types of generic parameters on types.");
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Reflection/SignatureTypes.netcoreapp.cs]: Why an explicit exception rather than Assert.False?

Comment (AtsushiKan) [reply src/System.Runtime/tests/System/Reflection/SignatureTypes.netcoreapp.cs]: No special reason - we don't have `Assert.Fail` and the choice between two slightly forced alternatives to it is pretty arbitrary in my mind.

Comment (stephentoub) [reply src/System.Runtime/tests/System/Reflection/SignatureTypes.netcoreapp.cs]: Why do we need Assert.Fail here?  I don't understand why this isn't just:
```C#
Assert.False(type.IsGenericTypeParameter);
Assert.True(type.IsGenericMethodParameter);
Assert.Equal("!!" + position, type.Name);
```

Comment (AtsushiKan) [reply src/System.Runtime/tests/System/Reflection/SignatureTypes.netcoreapp.cs]: In every place other than signature types, generic parameters come in two flavors and it looks weird to just assert that it's only one of the two without comment especially when it's mingled among four or five other assertions. So I made it clearer by calling out that special case and where we'd put the new asserts if we ever did support the creation of signture type generic parameters.

**Pull request merged**